Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,69 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
+    command = CommandsCommand.new
+    command.call
+  end
+
+  class CommandsCommand < Command
+    def initialize
+      super
+      desc "Show a list of built-in and external commands"
+      option "--quiet", "List only the names of commands without the header"
```

Comment:
`option, :quiet` makes more sense to me; we should assume long options. Also, think about how you'll handle any short options and options that only make sense if other options are applied.